public/components/data_connections/components/datasource.tsx

```diff
@@ -164,7 +164,11 @@ export function DataSource(props: any) {
                 icon={<EuiIcon size="xxl" type="bolt" />}
                 title={'Accelerate performance'}
                 description="Accelerate performance through OpenSearch indexing."
-                onClick={() => {}}
+                onClick={() =>
+                  window.location.assign(
+                    `${observabilityDatasourcesID}#/acceleration/${dataSource}`
```

Comment:
why do you need the datasourcesId here?

Reply:
Yes it wil be. But window assign directly replaces the route parent `localhost:9200/app/datasources#/acc/..` to `localhost:9200/acc..` along with the suffix. Thus need to add the route parent here.

Reply:
you might want to navigate to url instead of explicit location.assign https://github.com/opensearch-project/OpenSearch-Dashboards/blob/a6af77d6422a3b54fdcbaea7b19b5b2d541508b0/src/core/public/application/types.ts#L728-L764

Reply:
@joshuali925 will need some help here. I tried to change code as below:
```
coreRefs.application?.navigateToUrl(`#/acceleration/${dataSource}`)
```
The URL shifts from `http://localhost:5601/app/datasources#/manage/my_spark1` to `http://localhost:5601/app/datasources#/acceleration/my_spark1` but doesn't load the acceleration page. I might be doing something wrong here.

Reply:
hm not sure, the code is going to call history.push https://github.com/opensearch-project/opensearch-dashboards/blob/0709333c641f1e2bcaf116eb7d5504b33afcd71d/src/core/public/application/application_service.tsx#L138

and it seems to be a known issue in react router https://stackoverflow.com/questions/42941708/react-history-push-is-updating-url-but-not-navigating-to-it-in-browser

i brought this up because it's consistent with core and handles some edge cases like cross app or cross site navigation, but if it doesn't work probably you can use location.assign or update location.hash

Reply:
Resorted to use window.location.hash